### PR TITLE
Paynote v2

### PR DESCRIPTION
**Pull request by annatraussnig**
_TODOs:_
- _maybe we need a better text for the thank you note_

Implement new paynote banners according to new specification:

- banner appear before and after the article's body, doesn't disturb the reading
- logged out users are shown a mix of "buynotes" (which still include a link to signup for trial) and "trynotes"
- the rate at which buynotes and trynotes are assigned can be configured via a variable
- trynotes allow to sign up for trial from within the banner
- in-trial users are shown buynotes (without trial signup link)
- iOS app users are shown a thank you note, without any call to action
- migrate negative colors to styleguide

### overall appearance
Screenshot 2019-10-31 at 13 29 01
Screenshot 2019-11-04 at 11 15 49

### buynote, signed out user
Screenshot 2019-10-31 at 13 30 14
Screenshot 2019-11-04 at 11 17 07

### trynote, trial signup workflow
Screenshot 2019-10-31 at 13 31 34
Screenshot 2019-10-31 at 13 31 38
Screenshot 2019-10-31 at 13 31 50
Screenshot 2019-10-31 at 13 32 15

### buynote, in-trial user
Screenshot 2019-10-31 at 13 33 07

### CODE REVIEW

File: components/Article/Progress/index.js [hunk 1]
```diff
@@ -307,7 +310,7 @@ class Progress extends Component {
 }
 
 Progress.propTypes = {
-  children: PropTypes.object,
+  children: PropTypes.array || PropTypes.object,
```
Comment (tpreusse) [inline components/Article/Progress/index.js]:
```suggestion
  children: PropTypes.node,
```

File: components/Article/Page.js [hunk 2]
```diff
@@ -593,24 +609,22 @@ class ArticlePage extends Component {
...
-                  variation={payNoteVariation}
-                  expanded={isAwayFromBottomBar} />
-              )}
               {this.state.showPdf &&
               <PdfOverlay
                 article={article}
                 onClose={this.togglePdf} />}
               <ArticleGallery article={article} show={!!router.query.gallery} ref={this.galleryRef}>
                 <ProgressComponent article={article}>
+                  {splitContent.title && (<div style={{ marginBottom: 20 }}>
```
Comment (tpreusse) [inline components/Article/Page.js]: need media query like title block:
https://github.com/orbiting/styleguide/blob/358cd6ace0b2dec7138a9e96e3fc7b32ae257f58/src/components/TitleBlock/index.js#L19-L24

File: components/Article/PayNote.js [hunk 3]
```diff
@@ -1,231 +1,238 @@
...
+    <PayNoteCta
+      darkMode={isBefore}
+      variation={variation}
+      position={position}
+      translator={translator}
+      isTrialContext={isTrialContext} />
+
+  return <div {...styles.banner} style={{ backgroundColor: isBefore ? colors.negative.primaryBg : colors.primaryBg }}>
+    <Center>
+      { !isBefore && (<div {...styles.brand}><BrandMark /></div>) }
```
Comment (tpreusse) [inline components/Article/PayNote.js]: rm, Brigitte does not like it :)

File: components/Trial/Form.js [hunk 4]
```diff
@@ -151,24 +167,33 @@ const Form = (props) => {
...
               value={email.value}
               error={email.dirty && email.error}
               dirty={email.dirty}
               disabled={signingIn}
+              icon={minimal && (
+                <MdArrowForward style={{ cursor: 'pointer' }} size={30} onClick={requestAccess} />
+              )}
               onChange={(_, value, shouldValidate) => handleEmail(value, shouldValidate)} />
-            <div style={{ marginTop: narrow ? 10 : 40 }}>
+            <div style={{ marginTop: (narrow && 10) || (minimal && '0') || 40 }}>
```
[the remark this answers is not in the document]
Comment (annatraussnig) [reply components/Trial/Form.js]: I agree with you on the `'0'`, however nesting ternary operators feels equally uncomfortable to me.

File: components/Article/PayNote.js [hunk 5]
```diff
@@ -34,198 +41,272 @@ const styles = {
...
-export const Before = compose(
-  withT,
-  graphql(query),
-  withInNativeApp
-)(({ t, me, data: { membershipStats }, inNativeIOSApp, variation, expanded }) => (
-  <WithoutMembership render={() => {
-    if (me && me.activeMembership) {
-      return null
-    }
+const initTranslator = (t, membershipStats) => (
```
[the remark this answers is not in the document]
Comment (annatraussnig) [reply components/Article/PayNote.js]: Maybe… I feel like we do the same thing 3 times and that cluttering the components' code with this logic over and over again doesn't do much for clarity either, as it distracts from the main concern.

File: components/Trial/Form.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (annatraussnig) [reply components/Trial/Form.js]: leaving this for now

**Pull request merged**